blossom/templates/website/generic_form.html

```diff
@@ -149,52 +138,75 @@ <h4>{{ subheader }}</h4>
                     label.innerText = fileName ? fileName : "browse";
                 });
             });
-            // from https://www.tjvantoll.com/2012/08/05/html5-form-validation-showing-all-error-messages/
-            const createAllErrors = function () {
-                let form = $(this)
-                let errorList = $(".errorMessages");
 
-                const showAllErrorMessages = function () {
-                    errorList.innerHTML = "";
+            // Adapted from https://www.tjvantoll.com/2012/08/05/html5-form-validation-showing-all-error-messages/
+            const forms = Array.from(document.getElementsByTagName("form"));
 
-                    // Find all invalid fields within the form.
-                    form.find(":invalid").each(function (index, node) {
+            forms.forEach(function (form) {
+                const errorLists = form.getElementsByClassName("errorMessages");
 
-                        // Find the field's corresponding label
-                        let label = $("label[for=" + node.id + "] "),
-                            // Opera incorrectly does not fill the validationMessage property.
-                            message = node.validationMessage || 'Invalid value.';
-
-                        // jquery won't add the same class multiple times, so we can have
-                        // addClass run for each loop.
-                        errorList.addClass("mt-3");
-                        errorList.append(
-                            '<div class="alert alert-danger" role="alert">\n' + label.html() + " " + message + '\n</div>'
-                        )
-                    });
-                };
+                if (errorLists.length > 0) {
+                    const errorList = errorLists[0];
 
-                // Support Safari
-                form.on("submit", function (event) {
-                    if (this.checkValidity && !this.checkValidity()) {
-                        $(this).find(":invalid").first().focus();
-                        event.preventDefault();
-                    }
-                });
+                    const showAllErrorMessages = function () {
+                        errorList.innerHTML = "";
```

Comment:
If I'm not mistaken, this can be made more efficient by calling `errorList.forEach(function(el) { errorList.removeChildNode(el); })` because it wouldn't have to re-render the DOM.

Reply:
Based on this https://stackoverflow.com/questions/3955229/remove-all-child-elements-of-a-dom-node-in-javascript can I change it to `errorList.textContent = "";` instead? Seems to be a good compromise between simplicity/readability and performance. Additionally I don't think performance is *that* important here, as this won't get invoked often. Still worth to think about though, of course!